aiida_mlip/calculations/train.py

```diff
@@ -0,0 +1,170 @@
+"""Class for training machine learning models."""
+
+from pathlib import Path
+
+from aiida.common import InputValidationError, datastructures
+import aiida.common.folders
+from aiida.engine import CalcJob, CalcJobProcessSpec
+import aiida.engine.processes
+from aiida.orm import Dict, FolderData, SinglefileData
+
+from aiida_mlip.data.config import JanusConfigfile
+from aiida_mlip.data.model import ModelData
+
+
+def validate_inputs(
+    inputs: dict, port_namespace: aiida.engine.processes.ports.PortNamespace
+):
+    """
+    Check if the inputs are valid.
+
+    Parameters
+    ----------
+    inputs : dict
+        The inputs dictionary.
+
+    port_namespace : `aiida.engine.processes.ports.PortNamespace`
+        An instance of aiida's `PortNameSpace`.
+
+    Raises
+    ------
+    ValueError
+        Error message if validation fails, None otherwise.
+    """
+    if "mlip_config" in port_namespace:
+        if "mlip_config" not in inputs:
+            raise InputValidationError("No config file given")
+        mlip_dict = inputs.mlip_config.as_dictionary()
+        list_of_keys = ["train_file", "valid_file", "test_file", "name"]
+        for key in list_of_keys:
+            if key not in mlip_dict:
+                raise InputValidationError(f"Mandatory key {key} not in config file")
+            # Check if the keys actually correspond to a path except name which is
+            # just the name to use for the output files
+            if key in mlip_dict and key != "name":
```

Comment:
```suggestion
            if key != "name":
```
It will have raised an error `if key not in mlip_dict`